depinject/appconfig/config.go

```diff
@@ -48,67 +58,142 @@ func WrapAny(config protoreflect.ProtoMessage) *anypb.Any {
+			return depinject.Error(fmt.Errorf("module is missing name"))
 		}
 
-		msgType, err := protoregistry.GlobalTypes.FindMessageByURL(module.Config.TypeUrl)
+		msgTypeUrl, err := module.GetTypeUrl()
 		if err != nil {
 			return depinject.Error(err)
 		}
 
-		modules, err := internal.ModulesByProtoMessageName()
-		if err != nil {
-			return depinject.Error(err)
+		var descriptor protoreflect.MessageDescriptor
+
+		var moduleV2 *appv1alpha1.ModuleConfig
+		var moduleV1 *appv2.ModuleConfig
+
+		if isProtov2 {
+			moduleV2 = module.(*appv1alpha1.ModuleConfig)
+		} else {
+			moduleV1 = module.(*appv2.ModuleConfig)
 		}
 
-		init, ok := modules[msgType.Descriptor().FullName()]
+		if isProtov2 {
+			msgType, err := protoregistry.GlobalTypes.FindMessageByURL(msgTypeUrl)
+			if err != nil {
+				return depinject.Error(err)
+			}
+			descriptor = msgType.Descriptor()
+		} else {
+			des, err := proto.GogoResolver.FindDescriptorByName(protoreflect.FullName(GetProtoV1MsgNameFromAnyTypeUrl(msgTypeUrl)))
+			if err != nil {
+				return depinject.Error(err)
+			}
+			descriptor = des.(protoreflect.MessageDescriptor)
+		}
+
+		init, ok := modules[descriptor.FullName()]
 		if !ok {
-			modDesc := proto.GetExtension(msgType.Descriptor().Options(), appv1alpha1.E_Module).(*appv1alpha1.ModuleDescriptor)
+			modDesc := protov2.GetExtension(descriptor.Options(), appv1alpha1.E_Module).(*appv1alpha1.ModuleDescriptor)
 			if modDesc == nil {
 				return depinject.Error(fmt.Errorf("no module registered for type URL %s and that protobuf type does not have the option %s\n\n%s",
-					module.Config.TypeUrl, appv1alpha1.E_Module.TypeDescriptor().FullName(), dumpRegisteredModules(modules)))
+					msgTypeUrl, appv1alpha1.E_Module.TypeDescriptor().FullName(), dumpRegisteredModules(modules)))
 			}
 
 			return depinject.Error(fmt.Errorf("no module registered for type URL %s, did you forget to import %s: find more information on how to make a module ready for app wiring: https://docs.cosmos.network/main/building-modules/depinject\n\n%s",
-				module.Config.TypeUrl, modDesc.GoImport, dumpRegisteredModules(modules)))
+				msgTypeUrl, modDesc.GoImport, dumpRegisteredModules(modules)))
 		}
 
-		config := init.ConfigProtoMessage.ProtoReflect().Type().New().Interface()
-		err = anypb.UnmarshalTo(module.Config, config, proto.UnmarshalOptions{})
-		if err != nil {
-			return depinject.Error(err)
-		}
+		if isProtov2 {
+			config := init.ConfigProtoMessage.(protov2.Message).ProtoReflect().Type().New().Interface()
+			err = anypb.UnmarshalTo(moduleV2.Config, config, protov2.UnmarshalOptions{})
+			if err != nil {
+				return depinject.Error(err)
+			}
+			opts = append(opts, depinject.Supply(config))
+			for _, binding := range moduleV2.GolangBindings {
+				opts = append(opts, depinject.BindInterfaceInModule(moduleV2.GetName(), binding.InterfaceType, binding.Implementation))
+			}
+		} else {
+			config := init.ConfigProtoMessage.(proto.Message)
+
+			anyCfg := &gogotypes.Any{
```

Comment:
cast from `gogoproto/types/any.Any` to `gogoproto/types.Any` then `UnmarshalAny`. 
This codes can be replaced by 
```go
codectypes.NewInterfaceRegistry().UnpackAny()
``` 
but will make dependencies bigger as cosmos-sdk import